src/svp/SoundVelocityProfile.hpp

```diff
@@ -119,9 +133,43 @@ class SoundVelocityProfile {
 
         return speeds;
     }
+    
+    /**Returns the sound speed gradient*/
+    std::vector<double> & getSoundSpeedGradient() {
+        if(gradient.size() != samples.size()) {
+            for (unsigned int k=0; k < samples.size()-1; k++){
+                gradient.push_back( (getSpeeds()[k+1]- getSpeeds()[k])/(getDepths()[k+1]- getDepths()[k]) );
+            }
+        }
+        
+        return gradient;
+    }
+    
+    /**Returns the layer index at specified depth*/
+    unsigned int getLayerIndexForDraft() {
```

Comment:
This is in reference to an outdated commit.
Pings now contain a transducerDepth field.

Consider a SVP that has 2 samples, it will have 3 layers:
1. One layer above the first sample (index of layer is 0)
2. One layer between the two samples (index of layer is 1)
3. One layer below the last sample (index of layer is 2)

If the index of the layer < svp.size()
We form a layer with the depth of the transducer and the speed at the transducer and the svp data at this layer index.

If index >= svp.size()
Then we do a last layer propagation (constant speed) using the speed at the transducer.